sync/src/relayer/compact_block_process.rs

```diff
@@ -23,6 +23,18 @@ pub struct CompactBlockProcess<'a, CS> {
     peer: PeerIndex,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum Status {
+    UnknownParent,
+    NotBetter,
+    AlreadyInFlight,
+    AlreadyPending,
+    HeaderVerifyFailed,
+    InflightBlocksReachLimit,
+    SendMissingIndexes,
+    NoMissingIndexes,
+}
+
```

Comment:
1. It is worth a more reasonable name. For example, `SendMissingIndexes` is not a "status", `HeaderVerifyFailed` is an error.
2. Many other places can share this notion. Maybe we should define it as a global type, and use it to help score-system.

Reply:
1. Naming is hard.
2. Agree. There is too much `Result<(), E>`, it's not good to read, test and debug.

Reply:
`Status` is the only name which I can think, any suggestion?

Reply:
The problem here is that we misuse `Result` here, since we want to ignore some errors. It's better to categorize errors like HTTP, which 4xx are errors caused by clients, and 5xx are caused by the server.

Reply:
These are not errors.